packages/hyperion-util/src/PersistentData.ts

```diff
@@ -15,50 +17,73 @@ function getStorage(storageName: 'sessionStorage' | 'localStorage'): Storage | n
     storage.removeItem(x);
     return storage;
   } catch (e) {
-    return null;
+    return {
+      getItem(_key) {
+        return null;
+      },
+      setItem(_key, _value) {
+      }
+    };
   }
 }
 
 const SessionStorage = getStorage('sessionStorage');
 const LocalStorage = getStorage('localStorage');
 
-export const SESSION_DATA_SAVE_INTERVAL = 100; //ms
+export const SESSION_DATA_SAVE_INTERVAL = 20; //ms
 
-class PersistentData<T> {
-  private static runner: TimedTrigger | null = null;
-  private static pending = new Set<PersistentData<any>>();
-  private schedule() {
-    if (!this.isPersisted()) {
-      return; // Not much to do!
-    }
+class Scheduler {
+  private runner: TimedTrigger | null = null;
+  private pending = new Set<PersistentData<any>>();
+  public schedule: (data: PersistentData<any>) => void = data => {
+    // First call we decide what state to go in, then move to new states
 
-    if (!PersistentData.runner) {
-      PersistentData.runner = new TimedTrigger(
+    const firstRun = (data: PersistentData<any>) => {
+      assert(!this.runner, "Invalid state! First call should not have runner");
+      const runner = this.runner = new TimedTrigger(
         () => {
-          for (const i of PersistentData.pending) {
+          for (const i of this.pending) {
             i.save();
           }
-          PersistentData.pending.clear();
-          PersistentData.runner = null;
+          this.pending.clear();
+          this.runner = null;
+          this.schedule = firstRun;
         },
         SESSION_DATA_SAVE_INTERVAL
       );
-    } else {
-      PersistentData.runner.delay(); // If not saved yet, postpone
-    }
-    PersistentData.pending.add(this);
+      if (typeof window === "object" && typeof window.addEventListener === 'function') {
```

Comment:
this looks fine - mostly just curious - what is the case you're defending against by checking for the presence of aEL before calling it? Is it a security boundary or can this code run in places where aEL doesn't exist?

Reply:
This code sits in a package that may be called outside of main thread (e.g. WebWorker). Although as of now, the storage we support are only available in the main thread, but the plan is to later also support `IndexDB`. So, Same way we check the storages, we also check if `window.*` is also available.